rust/agents/relayer/src/msg/pending_operation.rs

```diff
@@ -0,0 +1,175 @@
+use std::cmp::Ordering;
+use std::time::Instant;
+
+use async_trait::async_trait;
+use enum_dispatch::enum_dispatch;
+use eyre::Report;
+
+#[allow(unused_imports)] // required for enum_dispatch
+use super::pending_message::PendingMessage;
+
+/// A pending operation that will be run by the submitter and cause a
+/// transaction to be sent.
+#[async_trait]
+#[enum_dispatch]
+pub trait PendingOperation {
+    /// Prepare to run this operation. This will be called before every run and
+    /// will usually have a very short gap between it and the run call.
```

Comment:
What is the "run call"?

A sequence diagram might help explain things. IIUC it's validate, prepare, submit?

Suggest putting them in that order in the trait as well

Reply:
Ah nvm I see now it's prepare, submit, validate, which makes sense given the order in this trait

Reply:
Yeah, fixed my doc